integration-tests/security/src/test/java/io/quarkiverse/openapi/generator/it/security/AuthorizationHeaderApiKeyCanFilterWithoutDuplicateAuthorizationTest.java

```diff
@@ -0,0 +1,42 @@
+package io.quarkiverse.openapi.generator.it.security;
+
+import static com.github.tomakehurst.wiremock.client.WireMock.equalTo;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+
+import java.util.List;
+
+import jakarta.inject.Inject;
+
+import org.acme.openapi.foo.api.FooResourceApi;
+import org.acme.openapi.foo.model.FooDTO;
+import org.eclipse.microprofile.rest.client.inject.RestClient;
+import org.junit.jupiter.api.Test;
+
+import com.github.tomakehurst.wiremock.WireMockServer;
+import com.github.tomakehurst.wiremock.client.WireMock;
+
+import io.quarkus.test.common.QuarkusTestResource;
+import io.quarkus.test.junit.QuarkusTest;
+
+@QuarkusTestResource(WiremockFoo.class)
+@QuarkusTest
+public class AuthorizationHeaderApiKeyCanFilterWithoutDuplicateAuthorizationTest {
+
+    WireMockServer fooServer;
+
+    @RestClient
+    @Inject
+    FooResourceApi fooResourceApi;
+
+    @Test
+    public void testNoMultipleAuthorizationHeadersAreSent() {
+        List<FooDTO> foos = fooResourceApi.getFoosUsingGET("not the fooapikey",
+                123465L);
+        assertNotNull(foos);
+        fooServer.verify(WireMock.getRequestedFor(
+                        WireMock.urlEqualTo("/api/foo/v2.0/foo?something=123465"))
+                .withHeader("Authorization", equalTo("fooapikey"))
```

Comment:
We might need to check if the header is not duplicated here. Isn't it the reason behind this PR?

Reply:
I see! Alternatively, you could add a new test case using RestAssured and verify the number of headers in the response.

Or maybe you might have access to https://github.com/wiremock/wiremock/blob/master/src/main/java/com/github/tomakehurst/wiremock/matching/RequestPattern.java#L496?

Reply:
Yep. I'll 👀 into these.

Basically it's about the request headers, the response doesn't really matter.
My use case triggers a 400 Bad request given the multiple headers and I can't do the very dirty thing from https://github.com/nickstenning/nginx-multiauth

As you probably guessed, the "weird" OpenApi definition I can't change it, mine comes from a 3rd party service vendor.

It's weird, but still spec compliant afaik. One party has the right to define a required `Authorization` as a `HeaderParam`.

The "problem" is that it clashes with `quarkus.openapi-generator.fooopenapi_json.auth.JWT.api-key` : this prop is not mandatory :  yet `ApiKeyAuthenticationProvider.java` wants a key. Otherwise it'd get a null one and write the empty "Authorization: emptyness" header.

My `.putSingle` works because it is _shadowing_ either the null key, but or & also the not-null HeaderParam one : at last only one Authorization will remain : the one from the `application.properties`, fixing the use case. Unsure if it's the right call... Thoughts on this ?

- Shall the ApiKeyAuthenticationProvider write a warning or info in logs at buildtime or runtime if one forgets to declare `api-key` in .properties ?

- I was pondering about using a noop Provider which basically, provides nothing : zero headers, zero filtering, per https://github.com/quarkiverse/quarkus-openapi-generator#custom-register-providers-for-generated-api

- This noop would shadow the ApiKeyAuthenticationProvider.

Reply:
oh sure, it's the request we make, so yeah getting the headers from WireMock instance should solve it.

Reply:
> Shall the ApiKeyAuthenticationProvider write a warning or info in logs at buildtime or runtime if one forgets to declare api-key in .properties ?

Yes, we can since it's the single source of ApiKey at the moment. A WARN in runtime would make total sense.

> I was pondering about using a noop Provider which basically, provides nothing : zero headers, zero filtering

How is this would be useful? 

LOL now that I see that we already support custom providers, @hbelmiro. About the discussion at #290

Reply:
I found a way to count the headers, added a warn log. Pushed. Tests pass.

I was thinking about a custom noop Provider simply to ignore the registration of `@RegisterProvider(CompositeAuthenticationProvider.class)` at _codegen_. Because the composite is the one pluggin the `ApiKeyAuthenticationProvider`.

This would avoid the header duplication bug by... hiding it by omission : if no ApiKeyAuthenticationProvider is registered then no Authorization headers are sent :smile_cat: 

Ignoring the registration woul be hacky. I believe my latest commits are better & ok enough.